File: ios/RSSignatureCaptureTests/PPSSignatureViewTests.m
#import <XCTest/XCTest.h>
#import "PPSSignatureView.h"

@interface PPSSignatureViewTests : XCTestCase
@property (nonatomic, strong) PPSSignatureView *signatureView;
@end

@implementation PPSSignatureViewTests

- (void)setUp {
    [super setUp];
    self.signatureView = [[PPSSignatureView alloc] initWithFrame:CGRectMake(0, 0, 320, 480)];
}

- (void)tearDown {
    self.signatureView = nil;
    [super tearDown];
}

- (void)testInitWithFrame_returnsNonNil {
    XCTAssertNotNil(self.signatureView);
}

- (void)testInitWithFrame_setsCorrectFrame {
    CGRect expected = CGRectMake(0, 0, 320, 480);
    XCTAssertTrue(CGRectEqualToRect(self.signatureView.frame, expected),
                  @"Frame should be (0, 0, 320, 480)");
}

- (void)testInitialHasSignature_isNO {
    XCTAssertFalse(self.signatureView.hasSignature,
                   @"hasSignature should be NO initially");
}

- (void)testErase_resetsHasSignature {
    // Simulate having a signature by setting it directly via KVC
    [self.signatureView setValue:@YES forKey:@"hasSignature"];
    XCTAssertTrue(self.signatureView.hasSignature);

    [self.signatureView erase];
    XCTAssertFalse(self.signatureView.hasSignature,
                   @"hasSignature should be NO after erase");
}

- (void)testSetStrokeColor_storesColor {
    UIColor *red = [UIColor redColor];
    self.signatureView.strokeColor = red;
    XCTAssertEqualObjects(self.signatureView.strokeColor, red,
                          @"strokeColor should be red after setting");
}

- (void)testSetBackgroundColor_storesColor {
    UIColor *blue = [UIColor blueColor];
    [self.signatureView setBackgroundColor:blue];
    // backgroundColor is stored internally as _bgColor and also set via super
    XCTAssertEqualObjects(self.signatureView.backgroundColor, blue,
                          @"backgroundColor should be blue after setting");
}

- (void)testSignatureImage_returnsNilWhenNoSignature {
    UIImage *image = [self.signatureView signatureImage];
    XCTAssertNil(image,
                 @"signatureImage should return nil when hasSignature is NO");
}

@end

File: ios/RSSignatureCaptureTests/PPSSignatureViewTests.swift
import XCTest

class PPSSignatureViewTests: XCTestCase {

    var signatureView: PPSSignatureView!

    override func setUp() {
        super.setUp()
        signatureView = PPSSignatureView(frame: CGRect(x: 0, y: 0, width: 320, height: 480))
    }

    override func tearDown() {
        signatureView = nil
        super.tearDown()
    }

    func testInitWithFrame_returnsNonNil() {
        XCTAssertNotNil(signatureView)
    }

    func testInitWithFrame_setsCorrectFrame() {
        let expected = CGRect(x: 0, y: 0, width: 320, height: 480)
        XCTAssertTrue(signatureView.frame.equalTo(expected),
                      "Frame should be (0, 0, 320, 480)")
    }

    func testInitialHasSignature_isNO() {
        XCTAssertFalse(signatureView.hasSignature,
                       "hasSignature should be NO initially")
    }

    func testErase_resetsHasSignature() {
        signatureView.hasSignature = true
        XCTAssertTrue(signatureView.hasSignature)

        signatureView.erase()
        XCTAssertFalse(signatureView.hasSignature,
                       "hasSignature should be NO after erase")
    }

    func testSetStrokeColor_storesColor() {
        let red = UIColor.red
        signatureView.strokeColor = red
        XCTAssertEqual(signatureView.strokeColor, red,
                       "strokeColor should be red after setting")
    }

    func testSetBackgroundColor_storesColor() {
        let blue = UIColor.blue
        signatureView.backgroundColor = blue
        XCTAssertEqual(signatureView.backgroundColor, blue,
                       "backgroundColor should be blue after setting")
    }

    func testSignatureImage_returnsNilWhenNoSignature() {
        let image = signatureView.signatureImage()
        XCTAssertNil(image,
                     "signatureImage should return nil when hasSignature is NO")
    }
}
